pkg/migration/cleanup.go

```diff
@@ -0,0 +1,241 @@
+package migration
+
+import (
+	"context"
+
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (h *Handler) deleteComponents(cluster *corev1.StorageCluster) error {
+	if err := h.deleteStorkComponent(cluster); err != nil {
+		return err
+	}
+	if err := h.deleteAutopilotComponent(cluster); err != nil {
+		return err
+	}
+	if err := h.deleteCSIComponent(cluster); err != nil {
+		return err
+	}
+	if err := h.deletePVCControllerComponent(cluster); err != nil {
+		return err
+	}
+	if err := h.deleteMonitoringComponent(cluster); err != nil {
+		return err
+	}
+	if err := h.deletePortworxAPI(cluster); err != nil {
+		return err
+	}
+	if err := h.deletePortworxRbac(cluster); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *Handler) deletePortworxRbac(cluster *corev1.StorageCluster) error {
+	if err := h.deleteObject("node-role-binding", "", &rbacv1.ClusterRoleBinding{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("node-get-put-list-role", "", &rbacv1.ClusterRole{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role-binding", cluster.Namespace, &rbacv1.RoleBinding{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role", cluster.Namespace, &rbacv1.Role{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role-binding", "portworx", &rbacv1.RoleBinding{}, cluster); err != nil {
```

Comment:
namespace is always "portworx"?

Reply:
Yes, this is the default namespace we had for storing volume encryption secrets. Our spec generator still uses that namespace by default.